### PR TITLE
Return the responses of put and delete operations

### PR DESCRIPTION
**Pull request by boolean5**
Hello,

This PR addresses issues #135 and #406. Its purpose is to:

1. Make `put` and `delete` functions return an `rpc_pb2.PutResponse` and an `rpc_pb2.DeleteRangeResponse` respectively. These responses contain a `header` which includes the `revision`, `raft_term`, `cluster_id` and `member_id`. If the `prev_kv` keyword argument was set to `True`, they also contain the key-value pair as it was before modification in the case of `put` and the deleted key-value pair in the case of `delete.` Similar responses are returned by put and delete operations when they are performed within a transaction. This information was previously discarded by python-etcd3 and is very likely to be needed by users.

2. Enable the use of the `prev_kv` keyword argument.

Regards!

### CODE REVIEW

File: etcd3/client.py [hunk 1]
```diff
@@ -389,9 +398,7 @@ def replace(self, key, initial_value, new_value):
 
         return status
 
-    def _build_delete_request(self, key,
-                              range_end=None,
-                              prev_kv=None):
+    def _build_delete_request(self, key, range_end=None, prev_kv=None):
```
Comment (jd) [inline etcd3/client.py]: why this change?

File: etcd3/client.py [hunk 2]
```diff
@@ -329,15 +329,19 @@ def get_all(self, sort_order=None, sort_target='key'):
             for kv in range_response.kvs:
                 yield (kv.value, KVMetadata(kv, range_response.header))
 
-    def _build_put_request(self, key, value, lease=None):
+    def _build_put_request(self, key, value, lease=None, prev_kv=None):
```
Comment (jd) [inline etcd3/client.py]: Can you use `False` rather than `None` by default? That sounds more logical since it's a boolean.

File: etcd3/client.py [hunk 3]
```diff
@@ -329,15 +329,19 @@ def get_all(self, sort_order=None, sort_target='key'):
...
         put_request.lease = utils.lease_to_id(lease)
+
+        if prev_kv is not None:
+            put_request.prev_kv = prev_kv
+
         return put_request
 
     @_handle_errors
-    def put(self, key, value, lease=None):
+    def put(self, key, value, lease=None, prev_kv=None):
```
Comment (jd) [inline etcd3/client.py]: ditto here. if you use `False` it gets clearer you want a boolean.

**Comment by boolean5** [comment]
Thanks for the feedback. The requested changes have been made. Let me know if there is anything else that needs to be done before merging.

File: etcd3/client.py [hunk 4]
```diff
@@ -422,34 +429,36 @@ def replace(self, key, initial_value, new_value):
...
-        delete_response = self.kvstub.DeleteRange(
+        delete_request = self._build_delete_request(key, prev_kv=prev_kv)
+
+        return self.kvstub.DeleteRange(
             delete_request,
             self.timeout,
             credentials=self.call_credentials,
             metadata=self.metadata
         )
-        return delete_response.deleted >= 1
```
[the remark this answers is not in the document]
Comment (boolean5) [reply etcd3/client.py]: How about a `return_response` kwarg that will default to `False`?

[the remark this answers is not in the document]
Comment (jd) [reply etcd3/client.py]: Sounds good!

[the remark this answers is not in the document]
Comment (boolean5) [reply etcd3/client.py]: Done.
